src/ShellCheck/Analytics.hs

```diff
@@ -4583,6 +4623,45 @@ checkArrayValueUsedAsIndex params _ =
 
             _ -> Nothing
 
+prop_checkDisabledSetEInIf1  = verifyTree    checkDisabledSetEInIf "set -e; f(){:;}; x=$(f)"
+prop_checkDisabledSetEInIf2  = verifyNotTree checkDisabledSetEInIf "f(){:;}; x=$(f)"
+prop_checkDisabledSetEInIf3  = verifyNotTree checkDisabledSetEInIf "set -e; f(){:;}; x=$(set -e; f)"
+prop_checkDisabledSetEInIf4  = verifyTree    checkDisabledSetEInIf "set -e; f(){:;}; f || echo"
+prop_checkDisabledSetEInIf5  = verifyNotTree checkDisabledSetEInIf "set -e; f(){:;}; echo || f"
+prop_checkDisabledSetEInIf6  = verifyTree    checkDisabledSetEInIf "set -e; f(){:;}; baz=$(set -e; f) || :"
+prop_checkDisabledSetEInIf7  = verifyNotTree checkDisabledSetEInIf "set -e; f(){:;}; baz=$(echo \"\") || :"
+prop_checkDisabledSetEInIf8  = verifyTree    checkDisabledSetEInIf "set -e; f(){:;}; f && echo"
+prop_checkDisabledSetEInIf9  = verifyTree    checkDisabledSetEInIf "set -e; f(){:;}; f || echo"
+prop_checkDisabledSetEInIf10 = verifyNotTree checkDisabledSetEInIf "set -e; f(){:;}; echo && f"
+prop_checkDisabledSetEInIf11 = verifyNotTree checkDisabledSetEInIf "set -e; f(){:;}; echo || f"
+prop_checkDisabledSetEInIf12 = verifyTree    checkDisabledSetEInIf "set -e; f(){:;}; ! f"
+prop_checkDisabledSetEInIf13 = verifyTree    checkDisabledSetEInIf "set -e; f(){:;}; if f; then :; fi"
+prop_checkDisabledSetEInIf14 = verifyTree    checkDisabledSetEInIf "set -e; f(){:;}; if set -e f; then :; fi"
+prop_checkDisabledSetEInIf15 = verifyTree    checkDisabledSetEInIf "set -e; f(){:;}; while f; do :; done"
+prop_checkDisabledSetEInIf16 = verifyTree    checkDisabledSetEInIf "set -e; f(){:;}; while set -e f; do :; done"
+prop_checkDisabledSetEInIf17 = verifyTree    checkDisabledSetEInIf "set -e; f(){:;}; until f; do :; done"
+prop_checkDisabledSetEInIf18 = verifyTree    checkDisabledSetEInIf "set -e; f(){:;}; until set -e f; do :; done"
+prop_checkDisabledSetEInIf19 = verifyTree    checkDisabledSetEInIf "set -e; f(){:;}; cat <(f)"
+prop_checkDisabledSetEInIf20 = verifyTree    checkDisabledSetEInIf "set -e; f(){:;}; cat <(set -e; f)"
+checkDisabledSetEInIf params t =
+    runNodeAnalysis checkCommand params t
+  where
+    checkCommand _ t@(T_SimpleCommand _ _ argv) =
+        when (hasSetE params && isSetESuppressed (getPath (parentMap params) t)) $
+            case getCommandNameAndToken False t of
+                (Just str, t) -> do
+                    let name = basename str
+                    mapM_ (checkArg name) argv
+                _ -> return ()
+    checkCommand _ _ = return ()
+
+    functions_ = functions t
+
+    checkArg cmd arg = sequence_ $ do
+        literalArg <- getUnquotedLiteral arg
+        definitionId <- Map.lookup literalArg functions_
+        return $ info (getId arg) 2304 "This function invocation will run with set -e disabled."
```

Comment:
You're not using `functions_`, `literalArg` or `definitionId` for anything here. The check you copied this code from had a message that included both the function definition and the function call site, that's why it used the function definition map, but this check's message only references the function call site.

Having both might be helpful if aliases are used, so that even if the call site uses an alias, the function definition can be highlighted to show why it was detected as a function alias and not a command alias.

Reply:
Ignore this, I forgot you need to call `Map.lookup` just to be able to tell if `arg` is a function call or a command call.

Reply:
I need to rewrite `checkArg` not to take an unused `cmd` parameter, so there's certainly still room for improvement here.